onnxruntime/core/providers/cuda/math/unary_elementwise_ops.cc

```diff
@@ -70,6 +70,56 @@ Status UnaryElementwise::Prepare(OpKernelContext* context, UnaryElementwisePrepa
                                                                                                   \
     return Status::OK();                                                                          \
   }
+#if !defined(DISABLE_FLOAT8_TYPES)
+#define OPSET20_CONSTRAINTS float, double, MLFloat16, BFloat16, Float8E4M3FN, Float8E4M3FNUZ, Float8E5M2, \
+                            Float8E5M2FNUZ
+#else
+#define OPSET20_CONSTRAINTS float, double, MLFloat16, BFloat16
+#endif
+
+ONNX_OPERATOR_VERSIONED_KERNEL_EX(
+    IsInf,
+    kOnnxDomain,
+    10,
+    19,
+    kCudaExecutionProvider,
+    (*KernelDefBuilder::Create())
+        .TypeConstraint("T1", BuildKernelDefConstraints<float, double>())
+        .TypeConstraint("T2", DataTypeImpl::GetTensorType<bool>()),
+    IsInf);
+
+ONNX_OPERATOR_KERNEL_EX(
+    IsInf,
+    kOnnxDomain,
+    20,
+    kCudaExecutionProvider,
+    (*KernelDefBuilder::Create())
+        .TypeConstraint("T1", BuildKernelDefConstraints<OPSET20_CONSTRAINTS>())
+        .TypeConstraint("T2", DataTypeImpl::GetTensorType<bool>()),
+    IsInf);
+
+IsInf::IsInf(const OpKernelInfo& info) : UnaryElementwise(info) {
+  int64_t detect_positive = 0, detect_negative = 0;
+  Status status = info.GetAttr("detect_positive", &detect_positive);
```

Comment:
spec: https://onnx.ai/onnx/operators/onnx__IsInf.html#isinf-20

detect_positive/detect_negative is optional, and its default value is 1.